uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/MediaPopupElementViewController.cpp

```diff
@@ -16,18 +16,81 @@
 #include "MediaPopupElementViewController.h"
 
 // Qt headers
-#include <QDebug>
+#include <QJsonObject>
+#include <QJsonDocument>
+#include <QJsonArray>
 
 // Maps SDK headers
 #include <MediaPopupElement.h>
 #include <PopupMedia.h>
 #include <PopupMediaListModel.h>
+#include <PopupTypes.h>
 
 // Toolkit headers
 #include <PopupMediaItem.h>
+#include <ImagePopupMediaItem.h>
+#include <BarChartPopupMediaItem.h>
+#include <PieChartPopupMediaItem.h>
+#include <LineChartPopupMediaItem.h>
 
 namespace Esri::ArcGISRuntime::Toolkit {
 
+namespace {
+
+  template <typename T>
+  bool isNullOrUndefined(const T& value) {
+      return value.isNull() || value.isUndefined();
+  }
+
+  // Get color information for each popup media. If no color is selected when defining a popup, GraphsTheme.Theme.MixSeries will
+  // be applied to the GraphsView.
+  // s.a. Web Map Spec: https://developers.arcgis.com/web-map-specification/objects/popupElement_media
+  QJsonArray colorsArrayFromJson(const QByteArray& mediaPopupElementJson, const int index)
+  {
+    if (mediaPopupElementJson.isEmpty())
+    {
+      qCritical() << QString("MediaPopupElement JSON is empty.");
```

Comment:
The helper verifies all the assumptions that are made with the json scheme as it currently is. It's thorough but I think it's necessary since if one of these parts do fail, that information would be helpful to user even if we handle the fail gracefully by passing forward an empty QJsonArray.